gapii/cc/vulkan_mid_execution.cpp

```diff
@@ -344,6 +349,19 @@ void VulkanSpy::serializeGPUBuffers(StateSerializer* serializer) {
     }
   }
 
+  // Retrieve buffers, one by one, chunk by chunk, to prevent OOM.
+  // Prepare one staging buffer of size kChunkSizeLimit per device.
+  std::map<VkDevice, std::shared_ptr<StagingBuffer>> staging_buffers;
```

Comment:
Why shared_ptr. I don't see any reason for needing a shared_ptr here.

Reply:
That's because StagingBuffer has a field which is a reference. How would you do it?

Reply:
std::unique_ptr.
shared_ptrs should be limited where we (incredibly rarely) need shared ownership.

Reply:
done, creating core/cc/make_unique.h along the way.